brave-tests/src/test/java/brave/propagation/B3PropagationTest.java

```diff
@@ -48,4 +48,15 @@
     assertThat(result)
         .isEqualTo(SamplingFlags.NOT_SAMPLED);
   }
+
+  @Test public void extractTraceContext_malformed() {
+    MapEntry mapEntry = new MapEntry();
+    map.put("x-b3-traceid", "1"); // ok
```

Comment:
What about declaring more examples here on what happens by using a data provider? that would be more expressive in terms on which cases this will/won't fail. What do you think?